### ISSUE TITLE
How to train NER with CRF for any Transformer model

### ISSUE THREAD

**Issue by pvcastro** [allenai/allennlp#4225]
**System:**
 - OS: Linux-4.15.0-91-generic-x86_64-with-debian-buster-sid
 - Python version: 3.7.7
 - AllenNLP version: I installed from master
 - PyTorch version: 1.4.0

**Question**
Hi there! I currently use a NER model based on crf_tagger model, trained with Bert in 0.9.0 version. It uses specific bert token indexers and embedders, as well as a "bert_adam" optimizer with learning rate 0.0005. It performs really well!
I've been meaning to assess other transformers models, such as albert or electra, which are available in transformers-2.8.0. I saw [this](https://github.com/allenai/allennlp-models/blob/master/training_config/nli/snli_roberta.jsonnet) example of an AllenNLP encoder using roberta, but in a way that uses the AutoModel from transformers, which would allow me to perform the evaluations I need. However, I'm having a hard time adapting my training_config to this new one, since some of these modules are no longer available in allennlp 1.0.0, such as bert_adam optimizer. Do you have any guidelines concerning training a new NER model based on crf_tagger and any given transformer model? What's the best training_config to use? The best optimizer? 

I've tried a few different combinations so far, but wasn't able to pass a 0.0 F1 in my benchmark. I have tried these:

`"optimizer": {
            "type": "adam",
            "lr": 0.001
        },`

`"optimizer": {
            "type": "huggingface_adamw",
            "lr": 2e-5,
            "weight_decay": 0.1,
        },`

`"optimizer": {
            "type": "huggingface_adamw",
            "lr": 0.0005
        },`

Besides, I wasn't able to use a pretrained_transformer token_indexer paired with a pretrained_transformer tokenizer, since conll2003 dataset reader doesn't support a tokenizer. So I'm using the mismatched version.

Here's the last training_config I used:

```
local transformer_model = "/home/models/albert/albert_checkpoint/";
local transformer_dim = 768;

{
    "dataset_reader": {
        "type": "conll2003",
        "coding_scheme": "BIOUL",
        "tag_label": "ner",
        "token_indexers": {
            "tokens": {
                "type": "pretrained_transformer_mismatched",
                "model_name": transformer_model,
                "max_length": 512
            }
        }
    },
    "train_data_path": "train.txt",
    "validation_data_path": "dev.txt",
    "test_data_path": "test.txt",
    "evaluate_on_test": true,
    "model": {
        "type": "crf_tagger",
        "calculate_span_f1": true,
        "constrain_crf_decoding": true,
        "dropout": 0.5,
        "encoder": {
            "type": "lstm",
            "bidirectional": true,
            "dropout": 0.5,
            "hidden_size": 200,
            "input_size": 768,
            "num_layers": 2
        },
        "include_start_end_transitions": false,
        "label_encoding": "BIOUL",
        "text_field_embedder": {
            "token_embedders": {
                "tokens": {
                    "type": "pretrained_transformer_mismatched",
                    "model_name": transformer_model,
                    "max_length": 512
                }
            }
        }
    },
    "data_loader": {
        "batch_sampler": {
            "type": "bucket",
            "batch_size" : 32,
            "sorting_keys": [
                "tokens"
            ]
        }
    },
    "trainer": {
        "cuda_device": 0,
        "num_epochs": 50,
        "optimizer": {
            "type": "adam",
            "lr": 0.001
        },
        "patience": 15,
        "checkpointer": {
            "num_serialized_models_to_keep": 1,
        },
        "validation_metric": "+f1-measure-overall"
    }
}
```

**Comment by schmmd** [comment]
@pvcastro we don't have any particular advice, but you might consider trying https://github.com/allenai/allentune
